workflow-dockstore-metrics/workflow_dockstore_metrics/pull_data.py

```diff
@@ -0,0 +1,180 @@
+"""
+This script pulls workflow execution data from BigQuery and sends it to Dockstore.
+It then updates the metadata table to indicate that the data has been sent to Dockstore.
+
+Requires:
+ - signing into a firecloud account with access to the broad-dsde-prod-analytics-dev project
+ - activating the associated poetry env with `poetry shell`, then poetry install
+
+Args:
+    -t --test: Run in test mode. Does not send metrics to dockstore or update the metadata table.
+    -l --login: Re-authenticate with gcloud
+"""
+
+from google.cloud import bigquery
+from google.oauth2 import service_account
+import requests
+import logging
+import os
+import argparse
+
+# configure logging format
+LOG_FORMAT = "%(asctime)s %(levelname)-8s %(message)s"
+LOG_LEVEL = "INFO"
+LOG_DATEFORMAT = "%Y-%m-%d %H:%M:%S"
+logging.basicConfig(
+    format=LOG_FORMAT,
+    level=getattr(logging, LOG_LEVEL),
+    datefmt=LOG_DATEFORMAT,
+)
+
+WORKFLOW_DATA_QUERY = '''
+WITH workflow_runtime_info AS ( 
+  SELECT  WORKFLOW_EXECUTION_UUID AS workflow_id, 
+          ARRAY_AGG(IF(METADATA_KEY = "status", METADATA_VALUE, NULL) IGNORE NULLS ORDER BY METADATA_TIMESTAMP DESC)[offset(0)] AS status, 
+          TIMESTAMP(ARRAY_AGG(IF(METADATA_KEY = "start", METADATA_VALUE, NULL) IGNORE NULLS ORDER BY METADATA_TIMESTAMP DESC)[offset(0)]) AS workflow_start, 
+          TIMESTAMP(ARRAY_AGG(IF(METADATA_KEY = "end", METADATA_VALUE, NULL) IGNORE NULLS ORDER BY METADATA_TIMESTAMP DESC)[offset(0)]) AS workflow_end, 
+          ARRAY_AGG(IF(METADATA_KEY = "submittedFiles:workflowUrl", METADATA_VALUE, null) IGNORE NULLS)[offset(0)] AS source_url 
+  FROM `broad-dsde-prod-analytics-dev.warehouse.cromwell_metadata` 
+  WHERE METADATA_TIMESTAMP > DATETIME_SUB(CURRENT_TIMESTAMP(), INTERVAL 2 DAY)
+  --WHERE SENT_TO_DOCKSTORE IS NULL
+  GROUP BY WORKFLOW_EXECUTION_UUID
+  HAVING STATUS != "Running"
```

Comment:
this is a new restriction, it doesn't make sense to send metrics for workflows that are still running